uikit/poidetail/src/main/java/com/sygic/ui/view/poidetail/viewmodel/PoiDetailInternalViewModel.kt

```diff
@@ -16,16 +21,19 @@ private val SHOWCASE_DELAY = TimeUnit.SECONDS.toMillis(1)
 const val DEFAULT_BEHAVIOR_STATE = BottomSheetBehavior.STATE_COLLAPSED
 const val SHOWCASE_BEHAVIOR_STATE = BottomSheetBehavior.STATE_EXPANDED
 
-internal class PoiDetailInternalViewModel(poiData: PoiData,
-                                 private val preferencesManager: PreferencesManager) : ViewModel(),
-    BottomSheetBehaviorWrapper.StateListener {
+internal class PoiDetailInternalViewModel(
+    data: ViewObjectData,
+    private val preferencesManager: PreferencesManager
+) : ViewModel(), BottomSheetBehaviorWrapper.StateListener {
 
-    val titleText: String
-    val subtitleText: String
-    val coordinatesText: String?
-    val urlText: String? = poiData.url
-    val emailText: String? = poiData.email
-    val phoneText: String? = poiData.phone
+    val coordinatesText: String? = data.position.getFormattedLocation().let { if (!it.isEmpty()) it else null }
```

Comment:
ma zmysel checkovat na empty? ak by sa tam priradilo `""` namiesto `null`, bol by to zasadny rozdiel?

Reply:
ano, v layoutu se checkuje na null a podle toho se nastavuje případně viditelnost.

Reply:
a nebolo by teda jednoduchsie checkovat v layoute na `empty`? ak je text prazdny, chces ho mat visible, alebo skryty ako pri `null`?

Reply:
jj, to bude lepší varianta ;)